src/test/scala/com/microsoft/hyperspace/index/DeltaLakeIntegrationTest.scala

```diff
@@ -0,0 +1,166 @@
+          }
+        }
+
+        assert(verifyIndexUsage(query().queryExecution.optimizedPlan, "deltaIndex"))
+
+        // Create a new version by deleting entries.
+        val deltaTable = DeltaTable.forPath(dataPath)
+        deltaTable.delete("clicks > 2000")
+
+        // The index should not be applied for the updated version.
+        assert(!verifyIndexUsage(query().queryExecution.optimizedPlan, "deltaIndex"))
+
+        // The index should be applied for the version at index creation.
+        assert(verifyIndexUsage(query(Some(0)).queryExecution.optimizedPlan, "deltaIndex"))
+
+        hyperspace.refreshIndex("deltaIndex")
+
+        // The index should be applied for the updated version.
+        assert(verifyIndexUsage(query().queryExecution.optimizedPlan, "deltaIndex/v__=1"))
+
+        // The index should not be applied for the version at index creation.
+        assert(!verifyIndexUsage(query(Some(0)).queryExecution.optimizedPlan, "deltaIndex"))
+      }
+    }
+  }
+
+  test("Verify Hybrid Scan on Delta Lake table.") {
+    withTempPathAsString { dataPath =>
+      import spark.implicits._
+      val dfFromSample = sampleData.toDF("Date", "RGUID", "Query", "imprs", "clicks")
+      dfFromSample.write.format("delta").save(dataPath)
+
+      val deltaDf = spark.read.format("delta").load(dataPath)
+      withSQLConf(IndexConstants.INDEX_LINEAGE_ENABLED -> "true") {
+        hyperspace.createIndex(deltaDf, IndexConfig("deltaIndex", Seq("clicks"), Seq("Query")))
+      }
+
+      withIndex("deltaIndex") {
+        def query(version: Option[Long] = None): DataFrame = {
+          if (version.isDefined) {
+            val deltaDf =
+              spark.read.format("delta").option("versionAsOf", version.get).load(dataPath)
+            deltaDf.filter(deltaDf("clicks") <= 2000).select(deltaDf("query"))
+          } else {
+            val deltaDf = spark.read.format("delta").load(dataPath)
+            deltaDf.filter(deltaDf("clicks") <= 2000).select(deltaDf("query"))
+          }
+        }
+
+        assert(verifyIndexUsage(query().queryExecution.optimizedPlan, "deltaIndex", false))
+
+        // Create a new version by deleting entries.
+        val deltaTable = DeltaTable.forPath(dataPath)
+        deltaTable.delete("clicks > 5000")
+
+        withSQLConf(
+          "spark.hyperspace.index.hybridscan.enabled" -> "true",
+          "spark.hyperspace.index.hybridscan.delete.enabled" -> "true") {
+          // The index should be applied for the updated version.
+          assert(verifyIndexUsage(query().queryExecution.optimizedPlan, "deltaIndex", true))
+
+          // Append data.
+          dfFromSample
+            .limit(3)
+            .write
+            .format("delta")
+            .mode("append")
+            .save(dataPath)
+
+          // The index should be applied for the updated version.
+          assert(verifyIndexUsage(query().queryExecution.optimizedPlan, "deltaIndex", true))
+        }
+      }
+    }
+  }
+
```

Comment:
Note to reviewers: more hybrid scan + delta lake test cases - https://github.com/microsoft/hyperspace/pull/274

Reply:
Is the result comparison tested in #274?

Reply:
yes `checkAnswer` is used to compare the results in #274